README.md

```diff
@@ -17,43 +17,40 @@ in the scripts folder and understand as much as possible. We'll touch some
 things briefly here that are better explained there. The steps here will expect
 you have some knowledge of what's going on inside the script.
 
-  1. Checkout LLVM to a to your home directory from
-     [OUR FORK](https://github.com/cmput415/llvm-project), checkout the 10.0.0
-     release, and change the directory in your script.
+  1. Checkout LLVM to your machine
      1. `cd $HOME`
-     1. `git clone git@github.com:cmput415/llvm-project.git`
-     1. `cd llvm`
-     1. `git checkout llvmorg-10.0.0`
-  1. Add these configuration lines to your `~/.bashrc` on linux or
-     `~/.bash_profile` on MacOS to setup the next steps. You should restart your
+     1. `git clone https://github.com/llvm/llvm-project.git`
+     1. `cd llvm-project`
+     1. `git checkout llvmorg-16.0.4`
+  1. Build MLIR
+     1. `mkdir build`
+     1. `cd build`
+     1. ```cmake -G Ninja ../llvm \
+             -DLLVM_ENABLE_PROJECTS=mlir \
+             -DLLVM_BUILD_EXAMPLES=ON \
+             -DLLVM_TARGETS_TO_BUILD="Native;NVPTX;AMDGPU" \
```

Comment:
You're compiling for GPU? Also interesting that `Native` is option, I had no idea.

Reply:
This is from the MLIR getting started page (which I should probably reference in the readme). I tested it on my Mac (which doesn't have an AMDGPU).

Reply:
> I tested it on my Mac (which doesn't have an AMDGPU).

This is how you get backends for cross compilling. Running with this would let your install produce code for your native architecture, Nvidia's PTX intermediate, and AMD GPUs (I don't know what this is), but wouldn't let you generate code for another backend such as PPC.

> A semicolon delimited list controlling which targets will be built and linked into llvm. The default list is defined as LLVM_ALL_TARGETS, and can be set to include out-of-tree targets. The default value includes: AArch64, AMDGPU, ARM, AVR, BPF, Hexagon, Lanai,
Mips, MSP430, NVPTX, PowerPC, RISCV, Sparc,
SystemZ, WebAssembly, X86, XCore. Setting this to "host" will only compile the host architecture (e.g. equivalent to specifying X86 on an x86 host machine) can significantly speed up compile and test times.

Apparently native is a synonym for host, so that should would, but I would definitely drop nvptx and amdgpu, I don't think anyone is targeting nvidia or amd gpus.

Reply:
Updated the build targets to "Native" only